src/lattice_reaction_system_diffusion.jl

```diff
@@ -0,0 +1,434 @@
+    end
+    function DiffusionReaction(rate::Num, species::Symbol)
+        new(rate, Symbolics.variable(species), species)
+    end
+    function DiffusionReaction(rate::Number, species::Symbol)
+        new(Num(rate), Symbolics.variable(species), species)
+    end
+    function DiffusionReaction(rate::Symbol, species::Symbol)
+        new(Symbolics.variable(rate), Symbolics.variable(species), species)
+    end
+end
+# Creates a vector of DiffusionReactions.
+function diffusion_reactions(diffusion_reactions)
+    [DiffusionReaction(dr[1], dr[2]) for dr in diffusion_reactions]
+end
+# Gets the parameters in a diffusion reaction.
+ModelingToolkit.parameters(dr::DiffusionReaction) = Symbolics.get_variables(dr.rate)
+
+### Lattice Reaction Network Structure ###
+# Desribes a spatial reaction network over a graph.
+struct LatticeReactionSystem # <: MT.AbstractTimeDependentSystem # Adding this part messes up show, disabling me from creating LRSs
+    """The reaction system within each comaprtment."""
+    rs::ReactionSystem
+    """The spatial reactions defined between individual nodes."""
+    spatial_reactions::Vector{<:AbstractSpatialReaction}
+    """The graph on which the lattice is defined."""
+    lattice::DiGraph
+
+    # Derrived values.
+    """The number of compartments."""
+    nC::Int64
+    """The number of edges."""
+    nE::Int64
+    """The number of species."""
+    nS::Int64
+    """Whenever the initial input was a di graph."""
+    init_digraph::Bool
+
+    function LatticeReactionSystem(rs::ReactionSystem,
+                                   spatial_reactions::Vector{<:AbstractSpatialReaction},
+                                   lattice::DiGraph; init_digraph = true)
+        return new(rs, spatial_reactions, lattice, nv(lattice), ne(lattice),
+                   length(species(rs)), init_digraph)
+    end
+    function LatticeReactionSystem(rs::ReactionSystem,
+                                   spatial_reactions::Vector{<:AbstractSpatialReaction},
+                                   lattice::SimpleGraph)
+        return LatticeReactionSystem(rs, spatial_reactions, graph_to_digraph(lattice);
+                                     init_digraph = false)
+    end
+    function LatticeReactionSystem(rs::ReactionSystem,
+                                   spatial_reaction::AbstractSpatialReaction,
+                                   lattice::Graphs.AbstractGraph)
+        return LatticeReactionSystem(rs, [spatial_reaction], lattice)
+    end
+end
+# Covnerts a graph to a digraph (in a way where we know where the new edges are in teh edge vector).
+function graph_to_digraph(g1)
+    g2 = Graphs.SimpleDiGraphFromIterator(reshape(permutedims(hcat(collect(edges(g1)),
+                                                       reverse.(edges(g1)))), :, 1)[:])
+    add_vertices!(g2, nv(g1) - nv(g2))
+    return g2
+end
+# Gets the species of a lattice reaction system.
+species(lrs::LatticeReactionSystem) = species(lrs.rs)
```

Comment:
Didn't think of this case. Would it make sense to allow people to define a species in a spatial reaction only though?

Reply:
I can imagine someone might want to just simulate transport to see what that gives for their graph before adding reactions.

Reply:
Yeah, that could be a thing, and it doesn't really hurt to support this anyway. Will extend and ensure that that works (and probably add a test to ensure that this can be done).

Reply:
Otherwise I guess the convention should be that the `rs` contains all species you might use, and only the transport parameter could be new? That also seems reasonable to me (and perhaps easier to manage).

Reply:
I was implementing it so that you could have species only appearing in the spatial reactions, but this caused some problems (i.e. you could have a system with 3 species but where the `ReactionSystem` has 2 species, however, the forcing and jacobian function of the non-spatial `ReactionSystem` are still used at places, which made things awkward when these were functions of less number of species than the system actually had). 

I might try to enable this in the future. In the meantime, there is a check that throws an error if a spatial reaction transports a species that is not declared in the non-spatial reaction system.